frontend/wallet/src/render/components/TransactionCell.re

```diff
@@ -209,46 +209,58 @@ module ActorName = {
 };
 
 module TimeDisplay = {
+  module Styles = {
+    open Css;
+
+    let time =
+      merge([
+        Theme.Text.Body.small,
+        style([
+          whiteSpace(`nowrap),
+          overflow(`hidden),
+          textOverflow(`ellipsis),
+          maxWidth(`rem(10.0)),
+        ]),
+      ]);
+  };
+
   [@react.component]
   let make = (~date: Js.Date.t) => {
     <span
       className=Css.(
-        merge([
-          Theme.Text.Body.small,
-          style([color(Theme.Colors.greyish(0.5))]),
-          style([
-            whiteSpace(`nowrap),
-            overflow(`hidden),
-            textOverflow(`ellipsis),
-            maxWidth(`rem(10.0)),
-          ]),
-        ])
+        merge([Styles.time, style([color(Theme.Colors.greyish(0.5))])])
       )>
       {ReasonReact.string(Time.render(~date, ~now=Js.Date.make()))}
     </span>;
   };
 };
 
 module Amount = {
+  module Styles = {
+    open Css;
+    let square = value =>
+      style([
+        Theme.Typeface.lucidaGrande,
+        color(value >= 0 ? Theme.Colors.serpentine : Theme.Colors.roseBud),
+      ]);
+
+    let currency =
+      style([color(Theme.Colors.greenblack), Theme.Typeface.plex]);
+  };
+
   [@react.component]
-  let make = (~decorated: bool, ~value: int) => {
-    <span className=Css.(style([alignSelf(`flexEnd)]))>
-      <span
-        className=Css.(
-          style([
-            Theme.Typeface.lucidaGrande,
-            color(
-              value >= 0 ? Theme.Colors.serpentine : Theme.Colors.roseBud,
-            ),
-          ])
-        )>
+  let make = (~value: int) => {
+    <>
+      <span className={Styles.square(value)}>
         {ReasonReact.string({j|■|j})}
       </span>
-      <span> {ReasonReact.string(Js.Int.toString(value))} </span>
-      {decorated
-         ? <span> {ReasonReact.string(value >= 0 ? "+" : "-")} </span>
-         : <span />}
-    </span>;
+      <span className=Styles.currency>
+        {ReasonReact.string(
+           " " ++ Js.Int.toString(value < 0 ? value * (-1) : value),
```

Comment:
Wow missed this one, will this even work? I was under the impression that all amounts would be positive and it's just the direction of the transaction that determines the sign. Can we avoid treating the value as an int? we should never be deserializing those values to ints (and can probably leave them as strings for almost all the UI)

Reply:
This would be a pretty significant change as these values are computed from transactions -- for example:

Here's one case:
```reasonml
Caml.List.exists(PublicKey.equal(from), myWallets)
            ? (-1) * amount - fee : amount,
```

Here's the coinbase case:
```reasonml
coinbase + transactionFees - proofFees - delegationFees,
```

I think when we figure out how we want to format our decimals we can add a constructor later from the raw coda integer `CurrencyFormatter.ofCodaInt(value)` or something.

Okay to leave for now?

Reply:
Basically the transaction payloads from the backend should be in raw ints I think